islands/CommentsList.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import { Signal, useSignal } from "@preact/signals";
+import { useEffect } from "preact/hooks";
+import { Comment } from "@/utils/db.ts";
+import CommentSummary from "@/components/CommentSummary.tsx";
+
+async function getComments(itemId: string, cursor: string) {
+  let url = `/api/items/${itemId}/comments`;
+  if (cursor !== "") url += "?cursor=" + cursor;
+  const resp = await fetch(url);
+  if (!resp.ok) throw new Error(`Request failed: GET ${url}`);
+  return await resp.json() as { comments: Comment[]; cursor: string };
+}
+
+export default function CommentsList(props: { itemId: string }) {
+  const commentsSig: Signal<Comment[]> = useSignal([]);
+  const cursorSig = useSignal("");
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      if (entry.isIntersecting) {
+        getComments(props.itemId, cursorSig.value)
+          .then(({ comments, cursor }) => {
+            if (cursor === "") observer.unobserve(entry.target);
+            commentsSig.value = [...commentsSig.value, ...comments];
+            cursorSig.value = cursor;
+          });
+      }
+    });
+
+    observer.observe(document.querySelector("#bottom")!);
+    return () => observer.disconnect();
+  }, []);
+
+  return (
+    <div>
+      {commentsSig.value.map((comment) => <CommentSummary {...comment} />)}
```

Comment:
Missing `key` prop on `CommentSummary`.

```suggestion
      {commentsSig.value.map((comment) => <CommentSummary key={comment.id} {...comment} />)}
```

Reply:
`<CommentSummary />` doesn't have a `key` property. Am I missing something?

Reply:
Every vnode has a `key` property. It's not forwarded to the component but instead a hint to the reconciler which element matches to what. It's required for rendering lists where the order of the elements can change. Without it Preact (or any other virtual-dom framework) won't be able to detect if elements have been swapped and will instead destroy and re-create them from scratch.

Reply:
I see! Added.

Reply:
1. Is this practice only good for client-side rendered components?
2. Perhaps, this should be a linting rule.